Formula/clap-host.rb

```diff
@@ -0,0 +1,24 @@
+class ClapHost < Formula
+  desc "CLAP example host"
+  homepage "https://github.com/free-audio/clap-host"
+  url "https://github.com/free-audio/clap-host", using: :git, tag: "1.0.1", revision: "4602d26b91a526ac80933bdc9f6fcafdf2423bbf"
+  license "MIT"
+  head "https://github.com/free-audio/clap-host", using: :git, branch: "main"
+
+  depends_on "catch2" => :build
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "qt"
+  depends_on "rtaudio"
+  depends_on "rtmidi"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DCLAP_HOST_BUNDLE=FALSE", *std_cmake_args
+    system "cmake", "--build", "build", "--target", "clap-host"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    system "clap-host", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This I'm afraid this will be complicated.
This application will require a CLAP plugin and will also try to open a GUI and access your audio interface and maybe a MIDI device.
I have unit tests in clap-host, but I don't install them and if I understand correctly test will test the installed files.

I'll think of something, thank you very much! 👍

Reply:
I've thought over it and I can't think anything useful that can be done for testing.
This is a GUI application, so it needs user interaction.
Also it is a host applications that requires a plugin to do anything useful, so unless there are clap plugins available on the system, it can't be tested.

Reply:
At least the `--help` ensures that the application starts and can resolves all its dependencies (dynamic libraries).

Reply:
Also I'd like to add that this application will be mostly used by plugin developers, so if something goes wrong they'll likely get back to us.

Reply:
There’s a bit of a chicken and an egg here. There are not CLAP plugins in homebrew yet because we don’t have the dev tools in peoples hands to do the ports etc…

You could do a test today by downloading the beta of surge-xt from https://surge-synthesizer.github.io/ and then running `clap-host -p “/Library/Audio/Plug-Ins/CLAP/Surge XT.clap”` which will work. Or I guess I could package up a test synth as a homebrew but then you would ask how do we test that. Which of course would require a host :)

But it would be really lovely to get clap-host into brew (and in a week or two clap-info and clap-val may be forthcoming).  The CLAP project has exploded in a way we didn’t expect and we are getting multiple ports a week happening. Having those devs be able to seamlessly access the test host through homebrew would be great!

Other than the two formatting changes that CI caught, any thoughts on how we could help get this over the finish line and into brew?

Much appreciated